### PR TITLE
thrift proxy: missing parts from the previous PR

### PR DESCRIPTION
**Pull request by rgs1**
These are leftovers from the previous PR:

* adds a counter for shadow requests that failed to be created
* cleanup part of UpstreamRequest's interface

Signed-off-by: Raul Gutierrez Segales <rgs@pinterest.com>

### CODE REVIEW

File: docs/root/configuration/other_protocols/thrift_filters/router_filter.rst [hunk 1]
```diff
@@ -23,6 +23,7 @@ The filter outputs generic routing error statistics in the *thrift.<stat_prefix>
   unknown_cluster, Counter, Total requests with a route that has an unknown cluster.
   upstream_rq_maintenance_mode, Counter, Total requests with a destination cluster in maintenance mode.
   no_healthy_upstream, Counter, Total requests with no healthy upstream endpoints available.
+  shadow_request_submit_failure, Counter, Total shadow requests that failed to be submitted
```
Comment (leyao-daily) [inline docs/root/configuration/other_protocols/thrift_filters/router_filter.rst:26]: to be submitted. (lack of ".") LOL

Comment (rgs1) [reply docs/root/configuration/other_protocols/thrift_filters/router_filter.rst]: Thanks :-)

**Review by zuercher — APPROVED** [review]
Looks good!

**Pull request merged**